### PR TITLE
Don't add ActionDispatch::Static middleware

### PR DESCRIPTION
**Pull request by danigonza**
### Summary

On the case that we want to have previews on production environment but rails is not managing the static files is not necessary to add the middleware `ActionDispatch::Static`. If not the an error saying `No such middleware to insert before: ActionDispatch::Static` will be raised when we executed `bundle exec rake assets:precompile`.

There is a configuration flag on rails that we can use to check if rails is serving static files: `public_file_server.enabled`. This flag on combination with `view_component.show_previews` is used to know if the middleware should be inserted or not

### CODE REVIEW

**Comment by primer-css** [comment]
:wave: Hello and thanks for pinging us! This issue or PR has been added to our inbox and a Design Infrastructure first responder will review it soon.
- :art: **If this is a PR that includes a visual change**, please make sure to add screenshots in the description or deploy this code to a lab machine with instructions for how to test.
- :fast_forward: **If this is a PR that includes changes to an interaction**, please include a [video recording](https://support.apple.com/en-us/HT208721) in the description.
- :warning: **If this is urgent**, please visit us in [#primer](https://github.slack.com/archives/CSGAVNZ19) on Slack and tag the first responders listed in the channel topic.

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ title: Changelog
 
 ## main
 
+* Don't add ActionDispatch::Static middleware, if rails does not serve static files
```
Comment (joelhawksley) [inline docs/CHANGELOG.md:10]:
```suggestion
* Don't add ActionDispatch::Static middleware unless `public_file_server.enabled`.
```

**Review by joelhawksley — APPROVED** [review]

**Review by camertron — APPROVED** [review]
Thank you!

**Review by joelhawksley — COMMENTED** [review]
@danigonza thanks for the PR! I think this is in place to support our usage of `prism` to highlight the source code of previews: https://github.com/ViewComponent/view_component/blob/c7db1481669dff519158c64d262bb8bbf211ccbe/app/views/view_components/_preview_source.html.erb#L1. What happens when viewing a preview with its source given the changes you're proposing here?

**Comment by joelhawksley** [comment]
@danigonza any interest in continuing work on this PR?

**Comment by reeganviljoen** [comment]
@joelhawksley since this seems inactive I created a new pull request to finish this off

**Comment by camertron** [comment]
Thanks for your work on this @danigonza. Closing in favor of https://github.com/ViewComponent/view_component/pull/1892

**Pull request closed without merging**